### PR TITLE
167891573 Setup JWT Helper 

### PR DESCRIPTION
**Pull request by femitj**
#### What does this PR do?
This PR sets up JSON web token helper

#### Description of Task to be completed?
 * Setup JSON web token
 * Setup bcrypt 

#### How should this be manually tested?
N/A

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
[#167891573](https://www.pivotaltracker.com/story/show/167891573)

#### Screenshots
N/A

#### Questions:
N/A

### CODE REVIEW

**Review by thatkazuk1 — COMMENTED** [review]
I thought we agreed on using ES6 classes @femitj

File: src/helpers/auth.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+const secret = process.env.SECRET;
+
+const Authenticate = {
+  /**
+   * @description - this method encodes a token
+   * @param {object} user
+   * @return {string} token
+   */
+  encode(user) {
+    const token = jwt.sign({ user }, secret, { expiresIn: '72h' });
```
Comment (max-wel) [inline src/helpers/auth.js]: Kindly make secret and expiresIn dynamic variables i.e function caller should be able to set those parameters

**Review by Pomile — COMMENTED** [review]
Nice Job

I think the expiry time for the token is too long ` expiresIn: '72h' `.  `24 hours` is ok.
Authenticate should be a service. Then the Object name `Authenticate` is not fit for this purpose. You can try `JwtHelper` instead of `Authenticate`. 

Also consider Es6 class. You could as well define the method to be static and call the method from where you want to use it without contructing the class.

File: src/helpers/auth.js (hunk 1 above)
Comment (femitj) [reply src/helpers/auth.js]: it will make sense for expiresIn but not for secret

**Comment by femitj** [comment]
> Nice Job
> 
> I think the expiry time for the token is too long `expiresIn: '72h'`. `24 hours` is ok.
> Authenticate should be a service. Then the Object name `Authenticate` is not fit for this purpose. You can try `JwtHelper` instead of `Authenticate`.
> 
> Also consider Es6 class. You could as well define the method to be static and call the method from where you want to use it without contructing the class.

you can call it whatever you want whenever you import it elsewhere, 'import jwtHelper from './helpers/auth'

alright es6 class noted, expires will be accordingly to what @sneakymaxy has mentioned

**Comment by femitj** [comment]
> I thought we agreed on using ES6 classes @femitj

this was after i raised a PR but noted and updated Desmond

**Comment by Pomile** [comment]
> > Nice Job
> > I think the expiry time for the token is too long `expiresIn: '72h'`. `24 hours` is ok.
> > Authenticate should be a service. Then the Object name `Authenticate` is not fit for this purpose. You can try `JwtHelper` instead of `Authenticate`.
> > Also consider Es6 class. You could as well define the method to be static and call the method from where you want to use it without contructing the class.
> 
> you can call it whatever you want whenever you import it elsewhere, 'import jwtHelper from './helpers/auth'
> 
> alright es6 class noted, expires will be accordingly to what @sneakymaxy has mentioned

I know you can call it whatever you like. I think it is better to give a distinct name that conveys a more meaningful message. This is to prevent confusion. Moreso it helps to describe the purpose of the object or function. 

Please let us maintain a linear naming convention so that our code can be more readable. (That is if you define JwtHelper as a class then it should be JwtHelper when i am calling it).

File: src/helpers/auth.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+/** Utils Class */
+class Utils {
+  /**
+   * @description - this method encodes a token
+   * @param {object} userObj
+   * @param {string} secret
+   * @param {string} expires
+   * @return {string} token
+   */
+  static encode(userObj, secret = envSecret, expires = '24h') {
```
Comment (thatkazuk1) [inline src/helpers/auth.js]: I see what you did there😆

**Comment by femitj** [comment]
> > > Nice Job
> > > I think the expiry time for the token is too long `expiresIn: '72h'`. `24 hours` is ok.
> > > Authenticate should be a service. Then the Object name `Authenticate` is not fit for this purpose. You can try `JwtHelper` instead of `Authenticate`.
> > > Also consider Es6 class. You could as well define the method to be static and call the method from where you want to use it without contructing the class.
> > 
> > 
> > you can call it whatever you want whenever you import it elsewhere, 'import jwtHelper from './helpers/auth'
> > alright es6 class noted, expires will be accordingly to what @sneakymaxy has mentioned
> 
> I know you can call it whatever you like. I think it is better to give a distinct name that conveys a more meaningful message. This is to prevent confusion. Moreso it helps to describe the purpose of the object or function.
> 
> Please let us maintain a linear naming convention so that our code can be more readable. (That is if you define JwtHelper as a class then it should be JwtHelper when i am calling it).

done!

**Review by max-wel — APPROVED** [review]

**Review by Pomile — APPROVED** [review]
Good job!

**Review by thatkazuk1 — APPROVED** [review]

**Review by max-wel — APPROVED** [review]

File: src/utils/helpers/auth.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
+import jwt from 'jsonwebtoken';
+import bcrypt from 'bcryptjs';
+import dotenv from 'dotenv';
+
+dotenv.config();
```
Comment (tvpeter) [inline src/utils/helpers/auth.js]: Have you tried taking out lines 3 and 5? I think it should work.

**Review by tvpeter — COMMENTED** [review]
Great job, just try the below suggestion.

**Review by tvpeter — APPROVED** [review]

**Review by daniellamarr — APPROVED** [review]

**Pull request merged**